src/test/java/org/broadinstitute/hellbender/engine/spark/ShuffleJoinReadsWithVariantsSparkUnitTest.java

```diff
@@ -17,7 +17,7 @@
 
 import java.util.*;
 
-public class JoinReadsWithVariantsSparkUnitTest extends BaseTest {
+public class ShuffleJoinReadsWithVariantsSparkUnitTest extends BaseTest {
```

Comment:
Could you rename this back to `JoinReadsWithVariantsSparkUnitTest`, have the `DataProvider` emit a `JoinStrategy.BROADCAST` and `JoinStrategy.SHUFFLE` argument for each existing test case, and have the test itself call either `ShuffleJoinReadsWithVariants.join()` or `BroadcastJoinReadsWithVariants.join()` depending on the `JoinStrategy` parameter?